Assets/HoloToolkit-Examples/UX/Scripts/InteractiveGroup.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace HoloToolkit.Examples.InteractiveElements
+{
+    [RequireComponent(typeof(InteractiveSet))]
+    public class InteractiveGroup : MonoBehaviour
+    {
+        [Tooltip("Prefab for your interactive.")]
+        public GameObject InteractivePrefab;
+
+        [Tooltip("Data to fill the InteractiveSet.")]
+        public List<string> Titles = new List<string>();
+
+        [Tooltip("number of elements after which we start a new column")]
+        public int Rows = 3;
+
+        public Vector2 Offsets = new Vector2(0.00f, 0.00f);
+
+        void Start()
+        {
+            Interactive interactive = InteractivePrefab.GetComponent<Interactive>();
+            if (interactive == null)
+            {
+                Debug.LogError("No interactive attached to Prefab, " +
+                    "please attach one of the Interactive scripts " +
+                    "to your InteractiveGroup Prefab!");
+            }
+            else
+            {
+                UpdateData();
+            }
+        }
+
+        private List<InteractiveToggle> Interactives {
+            get 
+            {
+                return GetInteractiveSet().Interactives;
+            }
+            set 
+            {
+                GetInteractiveSet().Interactives = value;
+            }
+        }
+
+        public InteractiveSet GetInteractiveSet()
+        {
+            return GetComponent<InteractiveSet>();
+        }
+
+        /// <summary>
+        /// create new Prefab-instance and fill with given data
+        /// </summary>
+        void CreateInteractives()
+        {
+            for (int i = Interactives.Count; i < Titles.Count; i++)
+            {
+                GameObject PrefabInst = Instantiate(InteractivePrefab, gameObject.transform) as GameObject;
+                InteractiveToggle InterInst = PrefabInst.GetComponent<InteractiveToggle>();
+                Interactives.Add(InterInst);
+            }
+        }
+
+        /// <summary>
+        /// shortcut to set title
+        /// (assuming this Interactive has a LabelTheme attached to it)
+        /// </summary>
+        /// <param name="title"></param>
+        public void SetTitle(string title)
+        {
+            LabelTheme lblTheme = gameObject.GetComponent<LabelTheme>();
+            if (lblTheme == null)
+            {
+                Debug.LogError("No LabelTheme attached to this Interactive");
```

Comment:
We could probably add a component requirement, and always get the component if null, and remove the debug.

Reply:
agree.